### Loading the Standard Libraries

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [3]:
data = pd.read_csv('zomato.csv',encoding= 'latin1')
data

Restaurant ID           Restaurant Name  Country Code              City  \
0           6317637          Le Petit Souffle           162       Makati City   
1           6304287          Izakaya Kikufuji           162       Makati City   
2           6300002    Heat - Edsa Shangri-La           162  Mandaluyong City   
3           6318506                      Ooma           162  Mandaluyong City   
4           6314302               Sambo Kojin           162  Mandaluyong City   
...             ...                       ...           ...               ...   
9546        5915730              NamlÛ± Gurme           208         ÛÁstanbul   
9547        5908749             Ceviz AÛôacÛ±           208         ÛÁstanbul   
9548        5915807                     Huqqa           208         ÛÁstanbul   
9549        5916112              Aôôk Kahve           208         ÛÁstanbul   
9550        5927402  Walter's Coffee Roastery           208         ÛÁstanbul   

                                                Address  \
0     Third Floor, Century City Mall, Kalayaan Avenu...   
1     Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2     Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3     Third Floor, Mega Fashion Hall, SM Megamall, O...   
4     Third Floor, Mega Atrium, SM Megamall, Ortigas...   
...                                                 ...   
9546  Kemankeô Karamustafa Paôa Mahallesi, RÛ±htÛ±...   
9547  Koôuyolu Mahallesi, Muhittin íìstí_ndaÛô Cadd...   
9548  Kuruí_eôme Mahallesi, Muallim Naci Caddesi, N...   
9549  Kuruí_eôme Mahallesi, Muallim Naci Caddesi, N...   
9550  CafeaÛôa Mahallesi, BademaltÛ± Sokak, No 21/B,...   

                                        Locality  \
0      Century City Mall, Poblacion, Makati City   
1     Little Tokyo, Legaspi Village, Makati City   
2     Edsa Shangri-La, Ortigas, Mandaluyong City   
3         SM Megamall, Ortigas, Mandaluyong City   
4         SM Megamall, Ortigas, Mandaluyong City   
...                                          ...   
9546                                    Karakí_y   
9547                                   Koôuyolu   
9548                                 Kuruí_eôme   
9549                                 Kuruí_eôme   
9550                                        Moda   

                                       Locality Verbose   Longitude  \
0     Century City Mall, Poblacion, Makati City, Mak...  121.027535   
1     Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101   
2     Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831   
3     SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475   
4     SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508   
...                                                 ...         ...   
9546                                Karakí_y, ÛÁstanbul   28.977392   
9547                               Koôuyolu, ÛÁstanbul   29.041297   
9548                             Kuruí_eôme, ÛÁstanbul   29.034640   
9549                             Kuruí_eôme, ÛÁstanbul   29.036019   
9550                                    Moda, ÛÁstanbul   29.026016   

       Latitude                          Cuisines  ...          Currency  \
0     14.565443        French, Japanese, Desserts  ...  Botswana Pula(P)   
1     14.553708                          Japanese  ...  Botswana Pula(P)   
2     14.581404  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)   
3     14.585318                   Japanese, Sushi  ...  Botswana Pula(P)   
4     14.584450                  Japanese, Korean  ...  Botswana Pula(P)   
...         ...                               ...  ...               ...   
9546  41.022793                           Turkish  ...  Turkish Lira(TL)   
9547  41.009847   World Cuisine, Patisserie, Cafe  ...  Turkish Lira(TL)   
9548  41.055817            Italian, World Cuisine  ...  Turkish Lira(TL)   
9549  41.057979                   Restaurant Cafe  ...  Turkish Lira(TL)   
9550  40.9

In [22]:
data.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   
3                   Japanese, Sushi  ...  Botswana Pula(P)                No   
4                  Japanese, Korean  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   
3                  No                No                   No           4   
4                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

#### Total columns present in the data

In [4]:
data.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')

### Total rows and columns in the data

In [5]:
data.shape

(9551, 21)

#### Observation:
         - Total numof rows 9551 and 21 columns in the data

### Deleting unnecessary columns in the data

In [23]:
col_to_del = ['Country Code', 'Address','Locality', 'Locality Verbose','Longitude', 'Latitude','Switch to order menu', 'Rating color']
col_to_del

['Country Code',
 'Address',
 'Locality',
 'Locality Verbose',
 'Longitude',
 'Latitude',
 'Switch to order menu',
 'Rating color']

In [25]:
updated_data = data.drop(col_to_del,axis = 1)
updated_data

Restaurant ID           Restaurant Name              City  \
0           6317637          Le Petit Souffle       Makati City   
1           6304287          Izakaya Kikufuji       Makati City   
2           6300002    Heat - Edsa Shangri-La  Mandaluyong City   
3           6318506                      Ooma  Mandaluyong City   
4           6314302               Sambo Kojin  Mandaluyong City   
...             ...                       ...               ...   
9546        5915730              NamlÛ± Gurme         ÛÁstanbul   
9547        5908749             Ceviz AÛôacÛ±         ÛÁstanbul   
9548        5915807                     Huqqa         ÛÁstanbul   
9549        5916112              Aôôk Kahve         ÛÁstanbul   
9550        5927402  Walter's Coffee Roastery         ÛÁstanbul   

                              Cuisines  Average Cost for two  \
0           French, Japanese, Desserts                  1100   
1                             Japanese                  1200   
2     Seafood, Asian, Filipino, Indian                  4000   
3                      Japanese, Sushi                  1500   
4                     Japanese, Korean                  1500   
...                                ...                   ...   
9546                           Turkish                    80   
9547   World Cuisine, Patisserie, Cafe                   105   
9548            Italian, World Cuisine                   170   
9549                   Restaurant Cafe                   120   
9550                              Cafe                    55   

              Currency Has Table booking Has Online delivery  \
0     Botswana Pula(P)               Yes                  No   
1     Botswana Pula(P)               Yes                  No   
2     Botswana Pula(P)               Yes                  No   
3     Botswana Pula(P)                No                  No   
4     Botswana Pula(P)               Yes                  No   
...                ...               ...                 ...   
9546  Turkish Lira(TL)                No                  No   
9547  Turkish Lira(TL)                No                  No   
9548  Turkish Lira(TL)                No                  No   
9549  Turkish Lira(TL)                No                  No   
9550  Turkish Lira(TL)                No                  No   

     Is delivering now  Price range  Aggregate rating Rating text  Votes  
0                   No            3               4.8   Excellent    314  
1                   No            3               4.5   Excellent    591  
2                   No            4               4.4   Very Good    270  
3                   No            4               4.9   Excellent    365  
4                   No            4               4.8   Excellent    229  
...                ...          ...               ...         ...    ...  
9546                No            3               4.1   Very Good    788  
9547                No            3               4.2   Very Good   1034  
9548                No            4               3.7        Good    661  
9549                No            4               4.0   Very Good    901  
9550                No            2               4.0   Very Good    591  

[9551 rows x 13 columns]

In [26]:
updated_data.head()

Restaurant ID         Restaurant Name              City  \
0        6317637        Le Petit Souffle       Makati City   
1        6304287        Izakaya Kikufuji       Makati City   
2        6300002  Heat - Edsa Shangri-La  Mandaluyong City   
3        6318506                    Ooma  Mandaluyong City   
4        6314302             Sambo Kojin  Mandaluyong City   

                           Cuisines  Average Cost for two          Currency  \
0        French, Japanese, Desserts                  1100  Botswana Pula(P)   
1                          Japanese                  1200  Botswana Pula(P)   
2  Seafood, Asian, Filipino, Indian                  4000  Botswana Pula(P)   
3                   Japanese, Sushi                  1500  Botswana Pula(P)   
4                  Japanese, Korean                  1500  Botswana Pula(P)   

  Has Table booking Has Online delivery Is delivering now  Price range  \
0               Yes                  No                No            3   
1               Yes                  No                No            3   
2               Yes                  No                No            4   
3                No                  No                No            4   
4               Yes                  No                No            4   

   Aggregate rating Rating text  Votes  
0               4.8   Excellent    314  
1               4.5   Excellent    591  
2               4.4   Very Good    270  
3               4.9   Excellent    365  
4               4.8   Excellent    229

In [27]:
updated_data.columns

Index(['Restaurant ID', 'Restaurant Name', 'City', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Price range',
       'Aggregate rating', 'Rating text', 'Votes'],
      dtype='object')

In [33]:
updated_data.shape

(9551, 13)

### Proportion of Excellent rating of restuarent

In [28]:
data['Rating text'].value_counts(normalize = True) * 100

Average      39.126793
Not rated    22.489792
Good         21.987226
Very Good    11.297246
Excellent     3.151502
Poor          1.947440
Name: Rating text, dtype: float64

In [35]:
updated_data[updated_data['Rating text'] == 'Excellent']

Restaurant ID                           Restaurant Name  \
0           6317637                          Le Petit Souffle   
1           6304287                          Izakaya Kikufuji   
3           6318506                                      Ooma   
4           6314302                               Sambo Kojin   
8           6300010  Spiral - Sofitel Philippine Plaza Manila   
...             ...                                       ...   
9537        5904116                       J'adore Chocolatier   
9538        5901782                                 Starbucks   
9540        5927248                          Draft Gastro Pub   
9543        5916085                              Dem Karakí_y   
9544        5915547                     Karakí_y Gí_llí_oÛôlu   

                  City                    Cuisines  Average Cost for two  \
0          Makati City  French, Japanese, Desserts                  1100   
1          Makati City                    Japanese                  1200   
3     Mandaluyong City             Japanese, Sushi                  1500   
4     Mandaluyong City            Japanese, Korean                  1500   
8           Pasay City     European, Asian, Indian                  6000   
...                ...                         ...                   ...   
9537         ÛÁstanbul                    Desserts                    50   
9538         ÛÁstanbul                        Cafe                    30   
9540         ÛÁstanbul                    Bar Food                   130   
9543         ÛÁstanbul                        Cafe                    35   
9544         ÛÁstanbul            Desserts, Bí_rek                    40   

              Currency Has Table booking Has Online delivery  \
0     Botswana Pula(P)               Yes                  No   
1     Botswana Pula(P)               Yes                  No   
3     Botswana Pula(P)                No                  No   
4     Botswana Pula(P)               Yes                  No   
8     Botswana Pula(P)               Yes                  No   
...                ...               ...                 ...   
9537  Turkish Lira(TL)                No                  No   
9538  Turkish Lira(TL)                No                  No   
9540  Turkish Lira(TL)                No                  No   
9543  Turkish Lira(TL)                No                  No   
9544  Turkish Lira(TL)                No                  No   

     Is delivering now  Price range  Aggregate rating Rating text  Votes  
0                   No            3               4.8   Excellent    314  
1                   No            3               4.5   Excellent    591  
3                   No            4               4.9   Excellent    365  
4                   No            4               4.8   Excellent    229  
8                   No            4               4.9   Excellent    621  
...                ...          ...               ...         ...    ...  
9537                No            2               4.7   Excellent   1311  
9538                No            2               4.9   Excellent   1042  
9540                No            4               4.9   Excellent    522  
9543                No            2               4.5   Excellent    761  
9544                No            2               4.7   Excellent   1305  

[301 rows x 13 columns]

### Observation:
- There is 39% of average ratings for the restaurants
- there is only 3% of restaurants having ratings Excellent
-11 % of restaurants having very good rating

### Total count of restuarants having aggregate rating greater than or equal to 4

In [38]:
 updated_data[updated_data[ 'Aggregate rating'] >= 4]['Restaurant Name'].count()

1380

### Analyzing city variable

In [40]:
updated_data['City'].value_counts()

New Delhi           5473
Gurgaon             1118
Noida               1080
Faridabad            251
Ghaziabad             25
                    ... 
Panchkula              1
Mc Millan              1
Mayfield               1
Macedon                1
Vineland Station       1
Name: City, Length: 141, dtype: int64

### Proportion of Text Ratings in New Delhi

In [41]:
updated_data[updated_data['City'] == 'New Delhi']['Rating text'].value_counts(normalize = True) * 100 # percentage

Average      45.587429
Not rated    26.036908
Good         20.610269
Very Good     5.481454
Poor          1.772337
Excellent     0.511602
Name: Rating text, dtype: float64

### Count of restuarents in New Delhi with aggregate rating >=4

In [44]:
updated_data[(updated_data['City'] == 'New Delhi') & (updated_data['Aggregate rating'] >=4)]['Restaurant Name'].count()

328

In [53]:
updated_data[['Restaurant Name','Cuisines']]

Restaurant Name                          Cuisines
0             Le Petit Souffle        French, Japanese, Desserts
1             Izakaya Kikufuji                          Japanese
2       Heat - Edsa Shangri-La  Seafood, Asian, Filipino, Indian
3                         Ooma                   Japanese, Sushi
4                  Sambo Kojin                  Japanese, Korean
...                        ...                               ...
9546              NamlÛ± Gurme                           Turkish
9547             Ceviz AÛôacÛ±   World Cuisine, Patisserie, Cafe
9548                     Huqqa            Italian, World Cuisine
9549              Aôôk Kahve                   Restaurant Cafe
9550  Walter's Coffee Roastery                              Cafe

[9551 rows x 2 columns]

In [61]:
data.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   
3                   Japanese, Sushi  ...  Botswana Pula(P)                No   
4                  Japanese, Korean  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   
3                  No                No                   No           4   
4                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

In [63]:
data.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')

In [64]:
col_del= ['Country Code', 'Address','Locality', 'Locality Verbose','Longitude', 'Latitude','Switch to order menu', 'Rating color']
col_del

['Country Code',
 'Address',
 'Locality',
 'Locality Verbose',
 'Longitude',
 'Latitude',
 'Switch to order menu',
 'Rating color']

In [66]:
data1 = data.drop(col_del,axis = 1)
data1

Restaurant ID           Restaurant Name              City  \
0           6317637          Le Petit Souffle       Makati City   
1           6304287          Izakaya Kikufuji       Makati City   
2           6300002    Heat - Edsa Shangri-La  Mandaluyong City   
3           6318506                      Ooma  Mandaluyong City   
4           6314302               Sambo Kojin  Mandaluyong City   
...             ...                       ...               ...   
9546        5915730              NamlÛ± Gurme         ÛÁstanbul   
9547        5908749             Ceviz AÛôacÛ±         ÛÁstanbul   
9548        5915807                     Huqqa         ÛÁstanbul   
9549        5916112              Aôôk Kahve         ÛÁstanbul   
9550        5927402  Walter's Coffee Roastery         ÛÁstanbul   

                              Cuisines  Average Cost for two  \
0           French, Japanese, Desserts                  1100   
1                             Japanese                  1200   
2     Seafood, Asian, Filipino, Indian                  4000   
3                      Japanese, Sushi                  1500   
4                     Japanese, Korean                  1500   
...                                ...                   ...   
9546                           Turkish                    80   
9547   World Cuisine, Patisserie, Cafe                   105   
9548            Italian, World Cuisine                   170   
9549                   Restaurant Cafe                   120   
9550                              Cafe                    55   

              Currency Has Table booking Has Online delivery  \
0     Botswana Pula(P)               Yes                  No   
1     Botswana Pula(P)               Yes                  No   
2     Botswana Pula(P)               Yes                  No   
3     Botswana Pula(P)                No                  No   
4     Botswana Pula(P)               Yes                  No   
...                ...               ...                 ...   
9546  Turkish Lira(TL)                No                  No   
9547  Turkish Lira(TL)                No                  No   
9548  Turkish Lira(TL)                No                  No   
9549  Turkish Lira(TL)                No                  No   
9550  Turkish Lira(TL)                No                  No   

     Is delivering now  Price range  Aggregate rating Rating text  Votes  
0                   No            3               4.8   Excellent    314  
1                   No            3               4.5   Excellent    591  
2                   No            4               4.4   Very Good    270  
3                   No            4               4.9   Excellent    365  
4                   No            4               4.8   Excellent    229  
...                ...          ...               ...         ...    ...  
9546                No            3               4.1   Very Good    788  
9547                No            3               4.2   Very Good   1034  
9548                No            4               3.7        Good    661  
9549                No            4               4.0   Very Good    901  
9550                No            2               4.0   Very Good    591  

[9551 rows x 13 columns]

In [68]:
data1['Currency'].value_counts()

Indian Rupees(Rs.)        8652
Dollar($)                  482
Pounds(Σ)                  80
Brazilian Real(R$)          60
Emirati Diram(AED)          60
Rand(R)                     60
NewZealand($)               40
Turkish Lira(TL)            34
Botswana Pula(P)            22
Indonesian Rupiah(IDR)      21
Qatari Rial(QR)             20
Sri Lankan Rupee(LKR)       20
Name: Currency, dtype: int64

In [97]:
zomato_india1 = data1[data1['Currency'] == 'Indian Rupees(Rs.)']
zomato_india1

Restaurant ID                        Restaurant Name   City  \
624         3400025                             Jahanpanah   Agra   
625         3400341                    Rangrezz Restaurant   Agra   
626         3400005                Time2Eat - Mama Chicken   Agra   
627         3400021  Chokho Jeeman Marwari Jain Bhojanalya   Agra   
628         3400017                         Pinch Of Spice   Agra   
...             ...                                    ...    ...   
9271        2800100                               D Cabana  Vizag   
9272        2800418                               Kaloreez  Vizag   
9273        2800881                                Plot 17  Vizag   
9274        2800042                       Vista - The Park  Vizag   
9275        2800019               Flying Spaghetti Monster  Vizag   

                                               Cuisines  Average Cost for two  \
624                               North Indian, Mughlai                   850   
625                               North Indian, Mughlai                   700   
626                                        North Indian                   500   
627                                          Rajasthani                   400   
628                      North Indian, Chinese, Mughlai                  1000   
...                                                 ...                   ...   
9271  Continental, Seafood, Chinese, North Indian, B...                   600   
9272                        Cafe, North Indian, Chinese                   400   
9273                             Burger, Pizza, Biryani                   600   
9274          American, North Indian, Thai, Continental                  1500   
9275                                            Italian                  1400   

                Currency Has Table booking Has Online delivery  \
624   Indian Rupees(Rs.)                No                  No   
625   Indian Rupees(Rs.)                No                  No   
626   Indian Rupees(Rs.)                No                  No   
627   Indian Rupees(Rs.)                No                  No   
628   Indian Rupees(Rs.)                No                  No   
...                  ...               ...                 ...   
9271  Indian Rupees(Rs.)                No                  No   
9272  Indian Rupees(Rs.)                No                  No   
9273  Indian Rupees(Rs.)                No                  No   
9274  Indian Rupees(Rs.)                No                  No   
9275  Indian Rupees(Rs.)                No                  No   

     Is delivering now  Price range  Aggregate rating Rating text  Votes  
624                 No            3               3.9        Good    140  
625                 No            2               3.5        Good     71  
626                 No            2               3.6        Good     94  
627                 No            2               4.0   Very Good     87  
628                 No            3               4.2   Very Good    177  
...                ...          ...               ...         ...    ...  
9271                No            2               3.6        Good    193  
9272                No            2               3.7        Good     85  
9273                No            2               4.3   Very Good    172  
9274                No            4               3.8        Good     74  
9275                No            3               4.4   Very Good    316  

[8652 rows x 13 columns]

## Total count of Cuisines in the Indian Restuarants

In [98]:
zomato_india1['Cuisines'].unique()

array(['North Indian, Mughlai', 'North Indian', 'Rajasthani', ...,
       'Continental, Seafood, Chinese, North Indian, Biryani',
       'Burger, Pizza, Biryani',
       'American, North Indian, Thai, Continental'], dtype=object)

In [100]:
zomato_india1['Cuisines'].count()

8652

## How Many restaurants have online delivery

In [105]:
zomato_india1.groupby('Has Online delivery').agg('count')

Restaurant ID  Restaurant Name  City  Cuisines  \
Has Online delivery                                                   
No                            6229             6229  6229      6229   
Yes                           2423             2423  2423      2423   

                     Average Cost for two  Currency  Has Table booking  \
Has Online delivery                                                      
No                                   6229      6229               6229   
Yes                                  2423      2423               2423   

                     Is delivering now  Price range  Aggregate rating  \
Has Online delivery                                                     
No                                6229         6229              6229   
Yes                               2423         2423              2423   

                     Rating text  Votes  
Has Online delivery                      
No                          6229   6229  
Yes                         2423   2423

In [108]:
zomato_india1.groupby('Has Online delivery')['Restaurant Name'].agg('count')

Has Online delivery
No     6229
Yes    2423
Name: Restaurant Name, dtype: int64

In [110]:
zomato_india1.groupby('Has Online delivery')['Restaurant Name'].agg('count') / len(zomato_india) * 100 # percentage

Has Online delivery
No     71.994914
Yes    28.005086
Name: Restaurant Name, dtype: float64

In [111]:
zomato_india1['City'].unique()

array(['Agra', 'Ahmedabad', 'Allahabad', 'Amritsar', 'Aurangabad',
       'Bangalore', 'Bhopal', 'Bhubaneshwar', 'Chandigarh', 'Chennai',
       'Coimbatore', 'Dehradun', 'Faridabad', 'Ghaziabad', 'Goa',
       'Gurgaon', 'Guwahati', 'Hyderabad', 'Indore', 'Jaipur', 'Kanpur',
       'Kochi', 'Kolkata', 'Lucknow', 'Ludhiana', 'Mangalore', 'Mohali',
       'Mumbai', 'Mysore', 'Nagpur', 'Nashik', 'New Delhi', 'Noida',
       'Panchkula', 'Patna', 'Puducherry', 'Pune', 'Ranchi',
       'Secunderabad', 'Surat', 'Vadodara', 'Varanasi', 'Vizag'],
      dtype=object)

## Maximum cost for two per city

In [115]:
zomato_india1.groupby('City')[['Restaurant Name','Average Cost for two']].agg('max')

Restaurant Name  Average Cost for two
City                                                              
Agra                 Time2Eat - Mama Chicken                  3600
Ahmedabad                     Yanki Sizzlers                  1800
Allahabad                  The Tamarind Tree                  1000
Amritsar                   The Yellow Chilli                  1200
Aurangabad                    d' Curry House                  1800
Bangalore                           Truffles                  2400
Bhopal                           Violet Hour                  1800
Bhubaneshwar                      Tyre Patty                  1500
Chandigarh                  Virgin Courtyard                  2200
Chennai                        Writer's Cafe                  2000
Coimbatore                    Zucca Pizzeria                  1500
Dehradun                 Y Cafe & Restaurant                  1600
Faridabad                         Yummy Cake                  1900
Ghaziabad                      The Dosa King                  2500
Goa                    The Fisherman's Wharf                  2000
Gurgaon                             iKitchen                  5000
Guwahati                                Ziya                  1650
Hyderabad           United Kitchens of India                  2000
Indore                          Waffle House                  2000
Jaipur         Zolocrust - Hotel Clarks Amer                  2500
Kanpur        Waterside - The Landmark Hotel                  3000
Kochi                            Tonico Cafe                  1500
Kolkata                            What's Up                  1900
Lucknow                      Vintage Machine                  2000
Ludhiana                   The Yellow Chilli                  2000
Mangalore                 Village Restaurant                  1600
Mohali                     The Shooters Cafe                   550
Mumbai                       The Rolling Pin                  1500
Mysore                       Vinayaka Mylari                  1700
Nagpur                               Zaikart                  1600
Nashik                    White X Sky Lounge                  1500
New Delhi               {Niche} - Cafe & Bar                  8000
Noida                        Zooby's Kitchen                  3500
Panchkula                      Hops n Grains                  2000
Patna                      The Yellow Chilli                  1600
Puducherry                     Zuka Choco-la                  1500
Pune                       The Urban Foundry                  2100
Ranchi                   Yo China Restaurant                  2000
Secunderabad                  Saffron Mantra                   850
Surat                          World Platter                  1500
Vadodara                         Wok On Fire                  1500
Varanasi           eastWEST - Radisson Hotel                  1100
Vizag                       Vista - The Park                  1700

In [117]:
zomato_india1.groupby('City')[['Restaurant Name','Average Cost for two']].agg(['max','min'])

Restaurant Name  \
                                         max   
City                                           
Agra                 Time2Eat - Mama Chicken   
Ahmedabad                     Yanki Sizzlers   
Allahabad                  The Tamarind Tree   
Amritsar                   The Yellow Chilli   
Aurangabad                    d' Curry House   
Bangalore                           Truffles   
Bhopal                           Violet Hour   
Bhubaneshwar                      Tyre Patty   
Chandigarh                  Virgin Courtyard   
Chennai                        Writer's Cafe   
Coimbatore                    Zucca Pizzeria   
Dehradun                 Y Cafe & Restaurant   
Faridabad                         Yummy Cake   
Ghaziabad                      The Dosa King   
Goa                    The Fisherman's Wharf   
Gurgaon                             iKitchen   
Guwahati                                Ziya   
Hyderabad           United Kitchens of India   
Indore                          Waffle House   
Jaipur         Zolocrust - Hotel Clarks Amer   
Kanpur        Waterside - The Landmark Hotel   
Kochi                            Tonico Cafe   
Kolkata                            What's Up   
Lucknow                      Vintage Machine   
Ludhiana                   The Yellow Chilli   
Mangalore                 Village Restaurant   
Mohali                     The Shooters Cafe   
Mumbai                       The Rolling Pin   
Mysore                       Vinayaka Mylari   
Nagpur                               Zaikart   
Nashik                    White X Sky Lounge   
New Delhi               {Niche} - Cafe & Bar   
Noida                        Zooby's Kitchen   
Panchkula                      Hops n Grains   
Patna                      The Yellow Chilli   
Puducherry                     Zuka Choco-la   
Pune                       The Urban Foundry   
Ranchi                   Yo China Restaurant   
Secunderabad                  Saffron Mantra   
Surat                          World Platter   
Vadodara                         Wok On Fire   
Varanasi           eastWEST - Radisson Hotel   
Vizag                       Vista - The Park   

                                                        Average Cost for two  \
                                                    min                  max   
City                                                                           
Agra                                       Bon Barbecue                 3600   
Ahmedabad                      650 - The Global Kitchen                 1800   
Allahabad                        Aryan Family's Delight                 1000   
Amritsar                             Ahuja Milk Bhandar                 1200   
Aurangabad                          Angeethi Restaurant                 1800   
Bangalore                     AB's - Absolute Barbecues                 2400   
Bhopal                                10 Downing Street                 1800   
Bhubaneshwar                        99 North Restaurant                 1500   
Chandigarh                              Barbeque Nation                 2200   
Chennai                       AB's - Absolute Barbecues                 2000   
Coimbatore                    24 Plus Cafe & Restaurant                 1500   
Dehradun                                        Anandam                 1600   
Faridabad                                   Aapki Rasoi                 1900   
Ghaziabad                                 9 Mars Lounge                 2500   
Goa                                             Antares                 2000   
Gurgaon                             19 Flavours Biryani                 5000   
Guwahati                        11th Avenue Cafe Bistro                 1650   
Hyderabad                     AB's - Absolute Barbecues                 2000   
Indore                                10 Downing Street                 2000   
Jaipur                                         Blackout     

In [118]:
zomato_india[zomato_india['Average Cost for two'] >= 8000]

Restaurant ID                    Restaurant Name       City Cuisines  \
7543           2701  Orient Express - Taj Palace Hotel  New Delhi      nan   

      Average Cost for two            Currency Has Table booking  \
7543                  8000  Indian Rupees(Rs.)               Yes   

     Has Online delivery Is delivering now  Price range  Aggregate rating  \
7543                  No                No            4               4.0   

     Rating text  Votes  
7543   Very Good    145